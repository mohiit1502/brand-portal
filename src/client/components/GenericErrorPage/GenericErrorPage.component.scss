.c-GenericErrorPage {
  margin-top: 8rem;
  color: #192340;

  .c-GenericErrorPage__image {
    margin-bottom: 2.8rem;
    transform: scale(1.1);
  }

  .c-GenericErrorPage__header {
    line-height: 1.8rem;
    margin-bottom: 0.8rem;
  }

  .c-GenericErrorPage__message {
    line-height: 1.2rem;
  }
}
