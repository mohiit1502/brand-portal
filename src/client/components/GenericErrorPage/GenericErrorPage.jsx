import React from 'react';
import PropTypes from 'prop-types';
import CONSTANTS from "../../constants/constants";
import * as images from "../../images";
import './GenericErrorPage.component.scss';

const GenericErrorPage = props => {
  return <div className={`${props.containerClass ? props.containerClass + " " : ""} mx-auto c-GenericErrorPage page-error text-center`} style={{maxWidth: "600px"}}>
    <img className="c-GenericErrorPage__image" src={images[props.image || "PageError"]} alt="Page Error" />
    {
      !props.generic ?
        <i>
          <h4 className="c-GenericErrorPage__header font-weight-bold">{props.header || "We’re sorry. An error has occurred."}</h4>
          <ol className="c-GenericErrorPage__message text-left d-inline-block">
            <li>
              If you are a new Brand Portal user, please try the following:
              <ul>
                <li>Visit <a href="/logout">Walmart Brand Portal</a></li>
                <li>Click “Register”</li>
                <li>Create a new account using a different email address than the one used for your Walmart supplier account</li>
              </ul>
            </li>
            <li>If you already have a Brand Portal account and you are seeing this error message, please reach out to us at <a href={`mailto:${CONSTANTS.LOGIN.CONTACTEMAIL}`}>{CONSTANTS.LOGIN.CONTACTEMAIL}</a></li>
          </ol>
        </i>
        :
        <React.Fragment>
          <h4 className="c-GenericErrorPage__header font-weight-bold">{props.header || "Oops. Something went wrong."}</h4>
          <p className="c-GenericErrorPage__message">{props.message || <span>Try to <a href={window.location.pathname}>refresh</a> this page or try again later.</span>}</p>
        </React.Fragment>
    }
    </div>;
};

GenericErrorPage.propTypes = {
  containerClass: PropTypes.string,
  generic: PropTypes.bool,
  image: PropTypes.string,
  header: PropTypes.string,
  message: PropTypes.string
};

export default GenericErrorPage;
