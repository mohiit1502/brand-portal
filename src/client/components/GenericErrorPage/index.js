import GenericErrorPage from "./GenericErrorPage.jsx";

export default GenericErrorPage;
