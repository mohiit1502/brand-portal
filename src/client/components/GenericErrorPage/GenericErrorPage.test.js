/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import GenericErrorPage from "./GenericErrorPage";

describe("GenericErrorPage", () => {
   it("renders without error", () => {

   });
});
