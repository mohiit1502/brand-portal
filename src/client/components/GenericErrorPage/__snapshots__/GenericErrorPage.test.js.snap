// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericErrorPage Tests renders without error with details 1`] = `
<div
  className="test-container  mx-auto c-GenericErrorPage page-error text-center"
  style={
    Object {
      "maxWidth": "600px",
    }
  }
>
  <img
    alt="Page Error"
    className="c-GenericErrorPage__image"
    src="page-error.svg"
  />
  <h4
    className="c-GenericErrorPage__header font-weight-bold"
  >
    generic-test-header
  </h4>
  <p
    className="c-GenericErrorPage__message"
  >
    test-message
  </p>
</div>
`;

exports[`GenericErrorPage Tests renders without error without details 1`] = `
<div
  className=" mx-auto c-GenericErrorPage page-error text-center"
  style={
    Object {
      "maxWidth": "600px",
    }
  }
>
  <img
    alt="Page Error"
    className="c-GenericErrorPage__image"
    src="page-error.svg"
  />
  <i>
    <h4
      className="c-GenericErrorPage__header font-weight-bold"
    >
      We’re sorry. An error has occurred.
    </h4>
    <ol
      className="c-GenericErrorPage__message text-left d-inline-block"
    >
      <li>
        If you are a new Brand Portal user, please try the following:
        <ul>
          <li>
            Visit 
            <a
              href="/logout"
            >
              Walmart Brand Portal
            </a>
          </li>
          <li>
            Click “Register”
          </li>
          <li>
            Create a new account using a different email address than the one used for your Walmart supplier account
          </li>
        </ul>
      </li>
      <li>
        If you already have a Brand Portal account and you are seeing this error message, please reach out to us at 
        <a
          href="mailto:IPInvest@walmart.com"
        >
          IPInvest@walmart.com
        </a>
      </li>
    </ol>
  </i>
</div>
`;
